sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1289,7 +1293,6 @@ type
     | INTERVAL
     | VARIANT
     | ARRAY | STRUCT | MAP
-    | unsupportedType=identifier
```

Comment:
otherwise, the following query 
```

CREATE TABLE t (a INT,
CONSTRAINT c1 CHECK (a > 0)
) USING parquet
```
will fail with
```
[UNSUPPORTED_DATATYPE] Unsupported data type "C1". SQLSTATE: 0A000
[info] == SQL (line 3, position 14) ==
[info]  CONSTRAINT c1 CHECK (a > 0)
```